bubble_sort_by.rb

```diff
@@ -0,0 +1,20 @@
+def bubble_sort_by(arr)
+  n = arr.length - 1
+  loop do
+    swapped = false
+    n.times do |i|
+      result = yield arr[i], arr[i + 1]
+      if result >= 1
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swapped = true
+      end
+    end
+    break if swapped == false
+  end
+end
```

Comment:
bubble_sort_by should return a sorted array. But I can not see the arr declaration after line no. 13. So place 'arr' after line 13 and you are ready to go.